siad/api_test.go

```diff
@@ -0,0 +1,340 @@
+}
+
+func reqWalletSend(t *testing.T, amount int, address string) SuccessResponse {
+	return reqSuccess(t, "/wallet/send?dest="+address+"&amount="+fmt.Sprintf("%d", amount))
+}
+
+func reqAddPeer(t *testing.T, peerAddr string) SuccessResponse {
+	return reqSuccess(t, "/peer/add?addr="+peerAddr)
+}
+
+func reqRemovePeer(t *testing.T, peerAddr string) SuccessResponse {
+	return reqSuccess(t, "/peer/remove?addr="+peerAddr)
+}
+
+// This might kill the program...
+func reqApplyUpdate(t *testing.T) SuccessResponse {
+	return reqSuccess(t, "/update/apply")
+}
+
+func reqCheckUpdate(t *testing.T) (Available bool, Version string) {
+	var response struct {
+		Version   string
+		Available bool
+	}
+	reqJSON(t, "/update/check", &response)
+	return response.Available, response.Version
+}
+
+func reqUploadFile(t *testing.T, localpath string, nickname string, filename string, pieces int) {
+	// Read in file
+	file, err := os.Open(localpath)
+	if err != nil {
+		t.Fatal("Couldn't open file to upload: " + err.Error())
+	}
+	defer file.Close()
+
+	// Write file to form
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	part, err := writer.CreateFormFile("file", filename)
+	if err != nil {
+		t.Fatal("Error creating form file: " + err.Error())
+	}
+	_, err = io.Copy(part, file)
+
+	// Populate other parameters
+	writer.WriteField("filename", filename)
+	writer.WriteField("nickname", nickname)
+	writer.WriteField("pieces", strconv.Itoa(pieces))
+
+	err = writer.Close()
+	if err != nil {
+		t.Fatal("Error closing writer: " + err.Error())
+	}
+
+	// Create the http request
+	req, err := http.NewRequest("POST", "http://127.0.0.1:9980/file/upload", body)
+	if err != nil {
+		t.Fatal("Error creating POST request: " + err.Error())
+	}
+	req.Header.Set("Content-Type", writer.FormDataContentType())
+
+	// Actually make the request
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Fatal("Error creating http client: " + err.Error())
+	}
+
+	// Read response body
+	resContent, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatal("Error reading response content: " + err.Error())
+	}
+
+	// Check the response
+	if res.StatusCode != http.StatusOK {
+		t.Fatal("/file/upload returne " + strconv.Itoa(res.StatusCode) + ": " + string(resContent))
+	}
+
+	//TODO: Read response into JSON success object
+	//TODO: return success object
+}
+
+func reqDownloadFile(t *testing.T, nickname string, filename string) SuccessResponse {
+	return reqSuccess(t, "/file/download?nickname="+nickname+"&filename="+filename)
+}
+
+func reqStop(t *testing.T) {
+	httpReq(t, "/stop")
+}
+
+func setupDaemon(t *testing.T) {
+
+	// Settings to speed up operations
+	consensus.BlockFrequency = consensus.Timestamp(1)
+	consensus.TargetWindow = consensus.BlockHeight(1000)
+	network.BootstrapPeers = []network.Address{"localhost:9988"}
+	consensus.RootTarget[0] = 255
+	consensus.MaxAdjustmentUp = big.NewRat(1005, 1000)
+	consensus.MaxAdjustmentDown = big.NewRat(995, 1000)
```

Comment:
Write a TODO here, this should all happen automatically when you use the makefile. It's a TODO for me, but it should be here nonetheless.